Comparisons fail when comparing a number with units to one without

I don't think this was the original behavior?  I think when we compared to a number without units before we made the quantity type that had units become dimensionless for the compare operation.

Currently it throws an exception:
```
2021-10-30 15:30:47.774 [TRACE] [nhab.core.library.types.QuantityType] - (Java::OrgOpenhabCoreLibraryTypes::QuantityType) 100 <=> 465.3 lx (Java::OrgOpenhabCoreLibraryTypes::QuantityType)
2021-10-30 15:30:47.775 [ERROR] [rulesupport.shared.simple.SimpleRule] - Can not compare incompatible units. (Java::JavaLang::IllegalArgumentException)
In rule: Turn on light due to motion
org.openhab.core.library.types.QuantityType.compareTo(org/openhab/core/library/types/QuantityType.java:245)
jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
jdk.internal.reflect.NativeMethodAccessorImpl.invoke(jdk/internal/reflect/NativeMethodAccessorImpl.java:62)
jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(jdk/internal/reflect/DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:566)
org.jruby.javasupport.JavaMethod.invokeDirectWithExceptionHandling(org/jruby/javasupport/JavaMethod.java:456)
org.jruby.javasupport.JavaMethod.invokeDirect(org/jruby/javasupport/JavaMethod.java:317)
RUBY.<=>(/etc/openhab/automation/lib/ruby/gem_home/gems/openhab-scripting-4.8.1/lib/openhab/dsl/types/quantity_type.rb:53)
org.jruby.RubyComparable.<(org/jruby/RubyComparable.java:207)
org.jruby.RubyComparable$INVOKER$s$1$0$op_lt.call(org/jruby/RubyComparable$INVOKER$s$1$0$op_lt.gen)
org.jruby.RubyNumeric.unwrapCoerced(org/jruby/RubyNumeric.java:629)
org.jruby.RubyNumeric.coerceRelOp(org/jruby/RubyNumeric.java:616)
org.jruby.RubyFixnum.op_ltOther(org/jruby/RubyFixnum.java:1142)
org.jruby.RubyFixnum.op_lt(org/jruby/RubyFixnum.java:1123)
org.jruby.RubyInteger$INVOKER$i$1$0$op_lt.call(org/jruby/RubyInteger$INVOKER$i$1$0$op_lt.gen)
org.jruby.RubyNumeric.unwrapCoerced(org/jruby/RubyNumeric.java:629)
org.jruby.RubyNumeric.coerceRelOp(org/jruby/RubyNumeric.java:616)
org.jruby.RubyFixnum.op_ltOther(org/jruby/RubyFixnum.java:1142)
org.jruby.RubyFixnum.op_lt(org/jruby/RubyFixnum.java:1123)
RUBY.<main>(/etc/openhab/automation/jsr223/ruby/personal/esp_home.rb:20)
org.jruby.RubyBasicObject.yieldUnder(org/jruby/RubyBasicObject.java:1778)
org.jruby.RubyBasicObject.instance_exec(org/jruby/RubyBasicObject.java:2694)
org.jruby.RubyBasicObject$INVOKER$i$0$3$instance_exec.call(org/jruby/RubyBasicObject$INVOKER$i$0$3$instance_exec.gen)
RUBY.process_trigger_task(/etc/openhab/automation/lib/ruby/gem_home/gems/openhab-scripting-4.8.1/lib/openhab/dsl/rules/automation_rule.rb:332)
RUBY.process_task(/etc/openhab/automation/lib/ruby/gem_home/gems/openhab-scripting-4.8.1/lib/openhab/dsl/rules/automation_rule.rb:290)
```

I think the old behavior is the appropriate one. If one items is missing units do a dimensionless compare. 
@ccutrer @jimtng ?

--- a/features/comparisons.feature
+++ b/features/comparisons.feature
@@ -24,6 +24,7 @@ Feature: comparisons
       | Number:Temperature | Number10C       | Number Ten              | Temperatures          | 10°C                      |
       | Number:Temperature | Number10C2      | Number Ten              | Temperatures          | 10°C                      |
       | Number:Temperature | Number20C       | Number Twenty           | Temperatures          | 20°C                      |
+      | Number:Illuminance | NumberLux       | Number Lux              |                       | 465.3 lx                  |
       | Dimmer             | Dimmer5         | Dimmer Five             |                       | 5                         |
       | Dimmer             | Dimmer10        | Dimmer Ten              |                       | 10                        |
       | Switch             | SwitchOne       | Switch One              | Switches              | ON                        |
@@ -178,6 +179,13 @@ Feature: comparisons
         [ Number10C                   , 'eql?', '10 °C'                     , false ]  ,
         [ Number10C                   , '>'   , '4 °F'                      , true  ]  ,
         [ Number10C                   , '=='  , '10 °C'                     , true  ]  ,
+
+        # NumberItem with UoM vs Integer
+        [ Number10C                   , 'eql?', 10                          , false ]  ,
+        [ Number10C                   , '>'   , 3                           , true  ]  ,
+        [ Number10C                   , '=='  , 10                          , true  ]  ,
+        [ NumberLux                   , '<'   , 100                         , false ]  ,
+        [ 100                         , '>'   , NumberLux                   , false ]  ,
 
         # NumberItem with UoM vs QuantityType
         [ Number10C                   , 'eql?', QuantityType.new('10°C')    , false ]  ,

--- a/lib/openhab/dsl/types/quantity_type.rb
+++ b/lib/openhab/dsl/types/quantity_type.rb
@@ -50,6 +50,9 @@ module OpenHAB
         def <=>(other) # rubocop:disable Metrics
           logger.trace("(#{self.class}) #{self} <=> #{other} (#{other.class})")
           if other.is_a?(self.class)
+            return unitize(other.unit).compare_to(other) if unit == ONE
+            return compare_to(other.unitize(unit)) if other.unit == ONE
+
             compare_to(other)
           elsif other.is_a?(Items::NumericItem) ||
                 (other.is_a?(Items::GroupItem) && other.base_item.is_a?(NumericItem))

